jenkins/e2e-image/e2e-runner.sh

```diff
@@ -251,18 +253,27 @@ function install_google_cloud_sdk_tarball() {
 }
 
 # Figures out the builtin k8s version of a GCI image.
+# If JENKINS_GCI_PATCH_K8S is set, returns the latest CI build on the same
+# branch instead.
 # Assumes: JENKINS_GCI_HEAD_IMAGE_FAMILY and KUBE_GCE_MASTER_IMAGE
 function get_gci_k8s_version() {
     if ! [[ "${JENKINS_USE_GCI_VERSION:-}" =~ ^[yY]$ ]]; then
         echo "JENKINS_USE_GCI_VERSION must be set."
         exit 1
     fi
     if [[ -z "${JENKINS_GCI_HEAD_IMAGE_FAMILY:-}" ]] || [[ -z "${KUBE_GCE_MASTER_IMAGE:-}" ]]; then
-        echo "JENKINS_GCI_HEAD_IMAGE_FAMILY and KUBE_GCE_MASTER_IMAGE must be set."
+        echo "JENKINS_GCI_HEAD_IMAGE_FAMILY and KUBE_GCE_MASTER_IMAGE must both be set."
         exit 1
     fi
-    local -r gci_k8s_version="v$(gsutil cat gs://container-vm-image-staging/k8s-version-map/${KUBE_GCE_MASTER_IMAGE})"
-    echo "${gci_k8s_version}"
+    local -r gci_k8s_version="$(gsutil cat gs://container-vm-image-staging/k8s-version-map/${KUBE_GCE_MASTER_IMAGE})"
+    if [[ "${JENKINS_GCI_PATCH_K8S}" =~ ^[yY]$ ]]; then
+      # We always want to test against the builtin k8s version, but occationally
+      # the builtin version has known bugs that keep our tests red. In those
+      # cases, we use the latest CI build on the same branch instead.
+      echo "ci/latest-${gci_k8s_version:0:3}"
```

Comment:
instead of changing `set_release_vars_from_gcs`, maybe make this side of the if branch call `set_release_vars_from_gcs "ci/latest-${gci_k8s_version:0:3}"`